sdk/how-to-guides/manage-storage.md

```diff
@@ -0,0 +1,232 @@
+    """Update the workload configuration."""
+    cache = self.model.storages["cache"]
+    if cache.location is None:
+        # This must be one of the other events. Return and wait for the storage-attached event.
+        logger.info("Storage is not yet ready.")
+        return
+    try:
+        self.push_configuration(cache_dir=cache.location)
+    except ops.pebble.ConnectionError:
+        # Pebble isn't ready yet. Return and wait for the pebble-ready event.
+        logger.info("Pebble is not yet ready.")
+        return
+```
+
+> Examples: [ZooKeeper ensuring that permission and ownership is correct](https://github.com/canonical/zookeeper-operator/blob/106f9c2cd9408a172b0e93f741d8c9f860c4c38e/src/charm.py#L247), [Kafka configuring additional storage](https://github.com/canonical/kafka-k8s-operator/blob/25cc5dd87bc2246c38fc511ac9c52f35f75f6513/src/charm.py#L298)
+
+<a href="#heading--observe-the-detaching-event-and-define-an-event-handler"><h3 id="heading--observe-the-detaching-event-and-define-an-event-handler">Observe the detaching event and define an event handler</h3></a>
+
+In the `src/charm.py` file, in the `__init__` function of your charm, set up an observer for the detacting event associated with your storage and pair that with an event handler. For example:
+
+```
+self.framework.observe(self.on.cache_storage_detaching, self._on_storage_detaching)
+```
+
+> See more: [StorageDetachingEvent](https://ops.readthedocs.io/en/latest/#ops.StorageDetachingEvent)
+
+Now, in the body of the charm definition, define the event handler, or adjust an existing holistic one. For example, to warn users that data won't be cached:
+
+```
+def _on_storage_detaching(self, event: ops.StorageDetachingEvent):
+    """Handle the storage being detached."""
+    self.unit.status = ops.ActiveStatus("Caching disabled; provide storage to boost performance)
+```
+
+> Examples: [MySQL handling cluster management](https://github.com/canonical/mysql-k8s-operator/blob/4c575b478b7ae2a28b09dde9cade2d3370dd4db6/src/charm.py#L823), [MongoDB updating the set before storage is removed](https://github.com/canonical/mongodb-operator/blob/b33d036173f47c68823e08a9f03189dc534d38dc/src/charm.py#L596)
+
+<a href="#heading--requesting-additional-storage"><h2 id="heading--requesting-additional-storage">Requesting additional storage</h2></a>
+
+If the charm needs additional units of a storage, it can request that with the `storages.request`
+method. The storage must be defined in the metadata as allowing multiple, for
+example:
+
+```yaml
+storage:
+    scratch:
+        type: filesystem
+        location: /scratch
+        multiple: 1-10
+```
+
+For example, if the charm needs to request two additional units of this storage:
+
+```python
+self.model.storages.request("scratch", 2)
+```
+
+The storage will not be available immediately after that call - the charm should
+observe the `storage-attached` event (as described above) and handle any remaining
```

Comment:
I'd avoid references of the form "as described above". They lead to entanglement.